Source/WebInspectorUI/UserInterface/Views/SpreadsheetRulesStyleDetailsPanel.js

```diff
@@ -90,8 +90,8 @@ WI.SpreadsheetRulesStyleDetailsPanel = class SpreadsheetRulesStyleDetailsPanel e
         }
 
         for (let section of this._sections) {
-            if (section.highlightProperty(property))
-                return;
+            section.deselectProperties();
+            section.highlightProperty(property);
```

Comment:
rather than call `deselectProperties` here i wonder if instead we should have `WI.SpreadsheetCSSStyleDeclarationEditor.prototype.highlightProperty` call this if nothing matches (i.e. right before the final `return false;`) as that would solve this bug and any like that also exist today or are accidentally added in the future

Reply:
This breaks when the old selection and new selection are under the same rule. I'd have to add it for both branches of `WI.SpreadsheetCSSStyleDeclarationEditor.prototype.highlightProperty` to work.

Also in the future there may be other features where the property is highlighted but the selection isn't invalidated.

Reply:
what exactly do you mean by "under the same rule"?  can you give an example?

Reply:
I.e. within the same CSS rule set

```css
.foo {
    color: blue;
    background: red;
}

.bar {
    border: 1px solid green;
}
```

If I select `color` and then click on the goto arrow for `border`, it will work, but `background` will still break